### ISSUE TITLE
Le jeu ne connait pas le fichier modifié

### ISSUE THREAD

**Issue by Yohan050** [Merlimocho/Eiyuden-Chronicle-FR#1]
Bonjour, j'ai essayé d'appliquer le patch comme indiqué sur Steam, mais le jeu ne reconnaît pas le nouveau fichier de traduction. C'est autant présent sur le menu principal que dans les textes du jeu. Ma version du jeu est la 1.0.5.

![image](https://github.com/Merlimocho/Eiyuden-Chronicle-FR/assets/29494032/72a80b51-4159-45c2-8c53-f5a34db1bd33)
![image](https://github.com/Merlimocho/Eiyuden-Chronicle-FR/assets/29494032/349f784a-02ea-4746-8b9e-70c8c09bf87f)

Au début, j'ai pensé que c'était parce que le fichier n'avait pas été compressé sur UABEA, car je voyais que le fichier faisait dans les 6 Mo alors que les autres fichiers faisaient un peu de 2 Mo. Je l'ai donc compressé dans le logiciel avec la méthode LZ4, mais le résultat est le même.
C'est peut-être une question de checksum, mais j'en doute.

Y a-t-il une solution ? Merci d'avance.

**Comment by Merlimocho** [comment]
Bonjour, 
J'ai essayé de remplacer le fichier de traduction de cette version par celui que j'utilise actuellement (avec quelques mises à jour supplémentaires). Peux-tu essayer de le télécharger de nouveau et me dire si le problème persiste s'il te plaît ?

**Comment by Yohan050** [comment]
Le problème est toujours là.

**Comment by Merlimocho** [comment]
J'avais eu ce problème au tout début lorsque je faisais encore des test avec UABEA, mon seul moyen de le contourner a été de réinstaller le jeu (ce qui peut être embêtant je le reconnais). 
Par contre sur la version actuelle, je n'ai aucun soucis. Je vais essayer de trouver ce qui cause le problème déjà. Tu as bien conservé le fichier original en backup ou tu veux que je te l'envoie ?

**Comment by Yohan050** [comment]
Je l'ai gardé sur mon bureau.

**Comment by Merlimocho** [comment]
As-tu essayé de : remettre le fichier original, relancer le jeu (en lançant une partie), pour ensuite remettre le fichier téléchargé ?

**Comment by Yohan050** [comment]
Je viens de tester. Quand je remets le fichier d'origine, le texte s'affiche. Mais dès que je remplace par le tien, j'ai de nouveau le No translation.

**Comment by Merlimocho** [comment]
Je n'arrive pas à comprendre dans quelles conditions recréer le problème...
J'essaye de voir ça de mon côté et je reviens vers toi dès que j'ai plus d'infos. Merci de l'avoir signalé en tout cas !

**Comment by Yohan050** [comment]
Pas de problème.

**Comment by Yohan050** [comment]
J'ai résolu le problème, du moins, je pense.

Il a l'air d'y avoir une mini différence entre la version steam et la version GOG sur laquelle je joue. 
Pour résoudre mon problème, j'ai extrait les textes de ton fichier avec uabea, j'ai fait l'import avec mon fichier chargé et ça a marché.

Il est possible qu'il y ait une différence d'en-tête entre les deux versions, même si, à ma connaissance, les écarts entre les magasins sont surtout au niveau de l'exécutable, avec ou sans drm, et de l'emplacement des dossiers de sauvegarde.

Serait-il possible d'avoir le CRC-32 de la version originale du fichier steam ? Celui de la version GOG est 550156FD.

Aussi, en comparant les fichiers dumpés avec un script que j'ai fait via chatgpt, hormis tes corrections de texte, ce qui est logique, j'ai trouvé qu'il y avait une ligne qui différait dans les deux dossier : 

```
Differences in b84e46ec8b54405a814f5ee4b7affc35.bundle-CAB-442776eb32297fcb9b3f9051c277637c-1.txt:
--- dump\b84e46ec8b54405a814f5ee4b7affc35.bundle-CAB-442776eb32297fcb9b3f9051c277637c-1.txt
+++ dump steam\b84e46ec8b54405a814f5ee4b7affc35.bundle-CAB-442776eb32297fcb9b3f9051c277637c-1.txt
@@ -40518,7 +40518,7 @@
   1 Array Array (2 items)
    0 int size = 2
    [0]
-    1 string data = "cab-18d0f95d2949db68e5ec8e4878887c30"
+    1 string data = "cab-8ff337dfcd3e31dee65a9cb2b1f3a4a8"
    [1]
     1 string data = "cab-ffa2f5012b840ee24c7b27879597cfdc"
  1 bool m_IsStreamedSceneAssetBundle = false
```

J'ignore si c'est de ton fait ou s'il s'agit d'une ligne qui varie selon les versions du jeu.

Je joins le script python que j'ai utilisé pour faire les comparaisons. Ça peut permettre d'aider à te rapporter des erreurs sur tes corrections sans avoir besoin de parcourir l'intégralité des fichiers. 
[https://pastebin.com/kiis74au](https://pastebin.com/kiis74au)

Je te mets aussi la copie de mon fichier de localisation.
[localization-string-tables-french(fr)_assets_all.zip](https://github.com/Merlimocho/Eiyuden-Chronicle-FR/files/15462769/localization-string-tables-french.fr._assets_all.zip)

J'espère que ça t'aidera. Au plaisir de ta réponse.

**Comment by Merlimocho** [comment]
Merci pour ta réponse parce que de mon côté j'en étais encore au point mort ne serait-ce que pour reproduire ton erreur. 

Voici le CRC-32 : 0410CD26,  cependant bien que ce soit le fichier original, mon backup était déjà décompressé, je ne sais pas si cela change quelque chose. 

J'ai malheureusement assez peu de connaissances dans ce domaine, uniquement de la bonne volonté, du coup je patauge un peu.  Par contre je m'interroge tout de même car j'ai déjà eu la même erreur que toi, lorsque j'avais fais quelques tests perso avant de faire le mod. Je pense pouvoir exclure la différence GOG / Steam. 

En regardant un petit peu : la ligne "cab-8ff337dfcd3e31dee65a9cb2b1f3a4a8" est présente dans le fichier de localisation français, anglais mais pas dans celui japonais. Je pensais que ça jouait sur la manière d'afficher les caractères mais on retrouve aussi la ligne en chinois simplifié par exemple. Je vais creuser là dessus. 

As-tu essayé de remplacer sur ma version modifiée, la ligne qui est différente en remettant la valeur de ta version originale ? 

Merci beaucoup pour le script et ton fichier de localisation ! Je vais continuer de regarder ça et de me renseigner de mon côté.

**Comment by Yohan050** [comment]
Oui, le fait que le fichier soit décompressé change le crc vu que les données changent d'organisation.
J'ai testé de changer la ligne qui différait, mais le problème de texte revient.
